docs/docs/components/guidelines/README.md

```diff
@@ -552,4 +762,823 @@ end user. Generate and use a GUID for the shared secret value, save it to a
+  processing of events: detecting new files and emitting events for those files
+  that meet certain criteria (for simplicity, the criteria here is that the type
+  of the file matches the expected ones):
+
+  ```javascript
+  const cloudStorage = require("../cloud_storage.app");
+
+  module.exports = {
+    props: {
+      cloudStorage,
+    },
+    methods: {
+      /**
+       * This method returns a boolean flag indicating whether a
+       * specific file is relevant and should be processed or not
+       *
+       * @param {object} file the file object included in the incoming
+       * event notification
+       * @returns {boolean} whether the file is relevant or not
+       */
+      isRelevant() {
+        throw new Error("isRelevant is not implemented");
+      },
+    },
+    async run(event) {
+      const { newFiles } = event;
+      const acceptedFileTypes = this.getFileTypes();
+      newFiles
+        .filter(this.isRelevant, this)
+        .forEach(this.$emit);
+    },
+  };
+  ```
+
+  You'll notice this implementation of the `isRelevant` method will in fact
+  result in an error. This is the equivalent of an abstract method but with a
+  runtime exception instead of a compile-time one. This means that:
+
+  - "Child" modules (i.e. modules that extend this one) must implement the
+    method depending on their particular use case
+  - The method should be documented so that other developers have an easier time
+    understanding its purpose and provide a concrete implementation that works
+
+  ::: warning
+  Note that this module is not an event source, but rather a module with code
+  to be shared by other modules.
+  :::
+
+- `cloud_storage/sources/new-file-of-type/new-file-of-type.js`
+
+  This is the new component, which is customizable via a user prop named
+  `fileTypes` (as suggested in the [Sensible Props](#sensible-props) section),
+  and extends the base component above to allow the user to freely select the
+  types of files to watch for:
+
+  ```javascript
+  const base = require("../common/new-file");
+
+  module.exports = {
+    ...base,
+    key: "cloud_storage-new-file-of-type",
+    name: "New File of Type",
+    description: "Emits an event whenever a file of a specific type is detected",
+    version: "0.0.1",
+    props: {
+      ...base.props,
+      fileTypes: {
+        type: "string[]",
+        label: "File Types",
+        description: "The types of files to watch for",
+        options: [
+          "application/pdf",
+          "application/zip",
+          "image/jpeg",
+          "image/png",
+          // Other types...
+        ],
+      },
+    },
+    methods: {
+      ...base.methods,
+      isRelevant({ type }) {
+        return this.fileTypes.includes(type);
+      },
+    },
+  };
+  ```
+
+- `cloud_storage/sources/new-pdf-document/new-pdf-document.js`
+
+  This component is a simplified version of the above, which also extends the
```

Comment:
Add a period after "module" and before "but".

Add a comma between "interest" and "it"